flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/typeutils/SymbolUtil.java

```diff
@@ -556,8 +595,10 @@ private static void addSymbolMapping(
     }
 
     private static void checkCalciteSymbol(Enum<?> calciteSymbol) {
+        final String className = calciteSymbol.getClass().getName();
         checkArgument(
-                calciteSymbol.getClass().getName().startsWith("org.apache.calcite."),
+                className.startsWith("org.apache.calcite.")
+                        || className.startsWith("com.google.common.collect."),
```

Comment:
why is that needed?

Reply:
I will add a comment to this. Unfortunately, for some very specific `RexNode` Calcite relies on Guava enums.